src/main/java/com/ezylang/evalex/data/EvaluationValue.java

```diff
@@ -374,7 +374,7 @@ public String getStringValue() {
   public Boolean getBooleanValue() {
     switch (getDataType()) {
       case NUMBER:
-        return !value.equals(BigDecimal.ZERO);
+        return ((BigDecimal) value).compareTo(BigDecimal.ZERO) != 0;
```

Comment:
I'd do it a little bit cleaner:
`return getNumberValue().compareTo(BigDecimal.ZERO) != 0;`

`getNumberValue()`already does the cast.